File: go/api/api_cloud_usage_submission.go
package api

import (
	"github.com/vayucode/vayu-sdks/go/client"
	"github.com/vayucode/vayu-sdks/go/openapi"
)

type CloudUsageResult = openapi.CloudUsageResult
type SubmitCloudUsageRequest = openapi.SubmitCloudUsageRequest

type CloudUsageSubmissionAPI struct {
	vayuClient *client.VayuClient
}

func NewCloudUsageSubmissionAPI(vayuClient *client.VayuClient) *CloudUsageSubmissionAPI {
	return &CloudUsageSubmissionAPI{vayuClient: vayuClient}
}

func NewSubmitCloudUsageRequest(results []CloudUsageResult, productCode string) *SubmitCloudUsageRequest {
	return &openapi.SubmitCloudUsageRequest{Results: results, ProductCode: productCode}
}

func (api *CloudUsageSubmissionAPI) SubmitCloudUsage(payload SubmitCloudUsageRequest) (map[string]interface{}, error) {
	ctx, cancel := client.GenerateContextWithTimeout()
	defer cancel()

	request := api.vayuClient.Client.CloudUsageSubmissionAPI.SubmitCloudUsage(ctx)
	request = request.SubmitCloudUsageRequest(payload)
	response, _, err := request.Execute()

	if err != nil {
		return nil, client.BuildVayuErrorFromGenericOpenAPIError(err)
	}

	return response, nil
}

File: python/vayu_sdk/apis/api_cloud_usage_submission.py
from openapi.api.cloud_usage_submission_api import CloudUsageSubmissionApi
from openapi.models.submit_cloud_usage_request import SubmitCloudUsageRequest
from vayu_sdk.clients.vayu_client import VayuClient


class CloudUsageSubmissionAPI:
    __client: CloudUsageSubmissionApi = None

    def __init__(self, vayu_client: VayuClient):
        self.__client = CloudUsageSubmissionApi(vayu_client.client)

    def submit(self, request: SubmitCloudUsageRequest):
        return self.__client.submit_cloud_usage(submit_cloud_usage_request=request)
